resources/js/components/Product/Price.vue

```diff
@@ -0,0 +1,139 @@
+            if (window.config.tax.calculation.price_includes_tax) {
+                if (displayTax) {
+                    return price
+                } else {
+                    return price / taxMultiplier
+                }
+            } else {
+                if (displayTax) {
+                    return price * taxMultiplier
+                } else {
+                    return price
+                }
+            }
+        },
+
+        getTaxPercent(product) {
+            let country_id = window.config.tax.defaults.country_id
+            let region_id = window.config.tax.defaults.region_id
+            let postcode = window.config.tax.defaults.postcode
+
+            if (['shipping', 'billing'].includes(window.config?.tax?.calculation.based_on)) {
+                country_id = window.app?.$data?.checkout?.[window.config?.tax?.calculation.based_on + '_address']?.country_id || country_id
+                region_id = window.app?.$data?.checkout?.[window.config?.tax?.calculation.based_on + '_address']?.region_id || region_id
+                postcode = window.app?.$data?.checkout?.[window.config?.tax?.calculation.based_on + '_address']?.postcode || postcode
+            }
+
+            let taxRate = window.config.tax.rates?.[product.tax_class_id]?.find(
+                (rate) =>
+                    rate.tax_country_id === country_id &&
+                    rate.tax_region_id * 1 === region_id * 1 &&
+                    postcode.match('^' + rate.tax_postcode.replace('*', '.*') + '$'),
+            )?.rate
+
+            if (taxRate === undefined || taxRate === null) {
+                console.debug('No tax rates found for', product, country_id, region_id, postcode)
+            }
+
+            return (taxRate ?? 0) / 100
+        },
+
+        calculateOptionsValue(basePrice, product, customOptions) {
+            return Object.entries(customOptions).reduce((priceAddition, [key, val]) => {
+                if (!val) {
+                    return priceAddition
+                }
+
+                let option = product.options.find((option) => option.option_id == key)
+                let optionPrice = ['drop_down', 'radio'].includes(option.type)
+                    ? option.values.find((value) => value.option_type_id == val).price
+                    : option.price
+
+                if (optionPrice.price_type == 'fixed') {
+                    return priceAddition * 1 + parseFloat(optionPrice.price)
+                }
+
+                return priceAddition * 1 + (parseFloat(basePrice) * parseFloat(optionPrice.price)) / 100
+            }, 0)
+        },
+    },
+
+    computed: {
+        specialPrice() {
+            JSON.stringify(this.options) // Hack: make vue recognize reactivity within the options object
+            return this.calculatePrice({ ...this.options, ...{ special_price: true } })
+        },
+
+        price() {
+            return this.calculatePrice(this.options)
+        },
```

Comment:
This needs to be thought out again, Magento doesn't work with prices this way.

Prices are calculated as follows: `Final Price=Min(Regular(Base) Price, Group(Tier) Price, Special Price, Catalog Price Rule) + Sum(Min Price per each required custom option)`

Note that prices are clamped to a minimum of 0 before and after the custom options are added. Also, the price gets rounded before getting multiplied by the qty.